client/components/map.jsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+
+export default class Map extends React.Component {
+  constructor(props) {
+    super(props);
+    this.mapRef = React.createRef();
+    this.mapInstance = null;
+    this.directionsService = null;
+    this.directionsRenderer = null;
+    this.calcRoute = this.calcRoute.bind(this);
```

Comment:
This method doesn't need to be bound, since it's not being passed as a callback anywhere.